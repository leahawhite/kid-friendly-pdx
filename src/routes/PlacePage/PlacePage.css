.place-page {
  width: 100%;
  margin: 0 auto;
}

.place-images-container {
  height: 100%;
  width: 100%;
  margin: 0px auto;
  padding: 10px;
}

@media only screen and (min-width: 950px) {
  .place-page {
    width: 80%;
  }
}

@media only screen and (min-width: 1200px) {
  .place-page {
    width: 70%;
  }
}

