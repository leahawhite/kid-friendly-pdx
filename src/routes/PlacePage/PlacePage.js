import React, { Component } from 'react';
import Carousel from '../../components/Carousel/Carousel';
import PlaceHeader from '../../components/PlaceHeader/PlaceHeader';
import PlaceActions from '../../components/PlaceActions/PlaceActions';
import PlaceMap from '../../components/PlaceMap/PlaceMap';
import PlaceHours from '../../components/PlaceHours/PlaceHours';
import PlaceFeatures from '../../components/PlaceFeatures/PlaceFeatures';
import PlaceReviews from '../../components/PlaceReviews/PlaceReviews';
import Spinner from '../../components/Spinner/Spinner';
import PlacesApiService from '../../services/places-api-service';
import './PlacePage.css';

export default class PlacePage extends Component {
  static defaultProps = {
    location: { state: {} },
    match: { params: {} },
    place: {},
    reviews: [],
  }
  constructor(props) {
    super(props)
    this.state = {
      place: {},
      reviews: [],
      isLoading: false,
      error: null,
    }
  }

  componentDidMount() {
    const { placeId } = this.props.match.params
    if (!this.props.location.state) {
      this.setState({isLoading: true})
      Promise.all([PlacesApiService.getById(placeId), PlacesApiService.getReviewsForPlace(placeId)])
      .then(([place, reviews]) => {
        this.setState({
          place: place,
          reviews: reviews,
          isLoading: false,
        })
      })
      .catch(error => {
        this.setState({ error: error })
      })
    } else {
      PlacesApiService.getReviewsForPlace(placeId)
      .then(reviews => {
        this.setState({
          reviews,
          isLoading: false,
        })
      })
      .catch(error => {
        this.setState({ error: error })
      })
    }
  }

  componentWillUnmount = () => {
    
  }
  
  render() {
    const { state } = this.props.location
    const place = state ? state.place : this.state.place
    const placeImages = place && place.images
    const { reviews } = this.state
    const { isLoading } = this.state
    if (isLoading) 
      return <Spinner />
      return (
        <div className="place-page">
          <section className="place-images-container">
            <Carousel images={placeImages} imagesClass="image-item" />
          </section>
          <PlaceHeader place={place} />
          <PlaceActions place={place} reviews={reviews} />
          <section className="place-addl-info">
            <PlaceMap place={place} />
            <PlaceHours place={place} />
            <PlaceFeatures place={place} />
            <PlaceReviews place={place} reviews={reviews} />
          </section>
        </div>
      )
  }
}


