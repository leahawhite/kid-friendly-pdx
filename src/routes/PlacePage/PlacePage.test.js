import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PlacePage from './PlacePage'

describe.skip('PlacePage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <PlacePage />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<PlacePage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
