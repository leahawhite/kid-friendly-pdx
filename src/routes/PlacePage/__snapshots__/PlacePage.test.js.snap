// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacePage component renders UI as expected 1`] = `
<div
  className="place-page"
>
  <section
    className="place-images-container"
  >
    <Carousel
      images={Array []}
      imagesClass="image-item"
    />
  </section>
  <PlaceHeader
    place={Object {}}
  />
  <PlaceActions
    place={Object {}}
  />
  <section
    className="place-addl-info"
  >
    <PlaceMap
      place={Object {}}
    />
    <PlaceHours
      place={Object {}}
    />
    <PlaceFeatures
      place={Object {}}
    />
    <PlaceReviews
      place={Object {}}
      reviews={Array []}
    />
  </section>
</div>
`;
